package defpackage;

import java.util.Collection;
import java.util.List;
import java.util.RandomAccess;

/* compiled from: r8-map-id-55bc5e14ca4f0ce1fb1e9e77cc19bf795986c6a69e3d49814dc3ca7768ce79b2 */
/* loaded from: /tmp/classes.dex */
public final class t40 implements RandomAccess {
    public Object[] e;
    public q40 f;
    public int g = 0;

    public t40(Object[] objArr) {
        this.e = objArr;
    }

    public final void a(int i, Object obj) {
        int i2 = this.g + 1;
        if (this.e.length < i2) {
            l(i2);
        }
        Object[] objArr = this.e;
        int i3 = this.g;
        if (i != i3) {
            System.arraycopy(objArr, i, objArr, i + 1, i3 - i);
        }
        objArr[i] = obj;
        this.g++;
    }

    public final void b(Object obj) {
        int i = this.g + 1;
        if (this.e.length < i) {
            l(i);
        }
        Object[] objArr = this.e;
        int i2 = this.g;
        objArr[i2] = obj;
        this.g = i2 + 1;
    }

    public final void c(int i, t40 t40Var) {
        int i2 = t40Var.g;
        if (i2 == 0) {
            return;
        }
        int i3 = this.g + i2;
        if (this.e.length < i3) {
            l(i3);
        }
        Object[] objArr = this.e;
        int i4 = this.g;
        if (i != i4) {
            System.arraycopy(objArr, i, objArr, i + i2, i4 - i);
        }
        System.arraycopy(t40Var.e, 0, objArr, i, i2);
        this.g += i2;
    }

    public final void d(int i, List list) {
        if (list.isEmpty()) {
            return;
        }
        int size = list.size();
        int i2 = this.g + size;
        if (this.e.length < i2) {
            l(i2);
        }
        Object[] objArr = this.e;
        int i3 = this.g;
        if (i != i3) {
            System.arraycopy(objArr, i, objArr, i + size, i3 - i);
        }
        int size2 = list.size();
        for (int i4 = 0; i4 < size2; i4++) {
            objArr[i + i4] = list.get(i4);
        }
        this.g += size;
    }

    public final boolean e(int i, Collection collection) {
        int i2 = 0;
        if (collection.isEmpty()) {
            return false;
        }
        int size = collection.size();
        int i3 = this.g + size;
        if (this.e.length < i3) {
            l(i3);
        }
        Object[] objArr = this.e;
        int i4 = this.g;
        if (i != i4) {
            System.arraycopy(objArr, i, objArr, i + size, i4 - i);
        }
        for (Object obj : collection) {
            int i5 = i2 + 1;
            if (i2 < 0) {
                throw new ArithmeticException("Index overflow has happened.");
            }
            objArr[i2 + i] = obj;
            i2 = i5;
        }
        this.g += size;
        return true;
    }

    public final List f() {
        q40 q40Var = this.f;
        if (q40Var != null) {
            return q40Var;
        }
        q40 q40Var2 = new q40(this);
        this.f = q40Var2;
        return q40Var2;
    }

    public final void g() {
        Object[] objArr = this.e;
        int i = this.g;
        for (int i2 = 0; i2 < i; i2++) {
            objArr[i2] = null;
        }
        this.g = 0;
    }

    public final boolean h(Object obj) {
        int i = this.g - 1;
        if (i >= 0) {
            for (int i2 = 0; !lw.i(this.e[i2], obj); i2++) {
                if (i2 != i) {
                }
            }
            return true;
        }
        return false;
    }

    public final boolean i(Object obj) {
        Object[] objArr = this.e;
        int i = this.g;
        int i2 = 0;
        while (true) {
            if (i2 >= i) {
                i2 = -1;
                break;
            }
            if (lw.i(obj, objArr[i2])) {
                break;
            }
            i2++;
        }
        if (i2 < 0) {
            return false;
        }
        j(i2);
        return true;
    }

    public final Object j(int i) {
        Object[] objArr = this.e;
        Object obj = objArr[i];
        int i2 = this.g;
        if (i != i2 - 1) {
            int i3 = i + 1;
            System.arraycopy(objArr, i3, objArr, i, i2 - i3);
        }
        int i4 = this.g - 1;
        this.g = i4;
        objArr[i4] = null;
        return obj;
    }

    public final void k(int i, int i2) {
        if (i2 > i) {
            int i3 = this.g;
            if (i2 < i3) {
                Object[] objArr = this.e;
                System.arraycopy(objArr, i2, objArr, i, i3 - i2);
            }
            int i4 = this.g;
            int i5 = i4 - (i2 - i);
            int i6 = i4 - 1;
            if (i5 <= i6) {
                int i7 = i5;
                while (true) {
                    this.e[i7] = null;
                    if (i7 == i6) {
                        break;
                    } else {
                        i7++;
                    }
                }
            }
            this.g = i5;
        }
    }

    public final void l(int i) {
        Object[] objArr = this.e;
        int length = objArr.length;
        Object[] objArr2 = new Object[Math.max(i, length * 2)];
        System.arraycopy(objArr, 0, objArr2, 0, length);
        this.e = objArr2;
    }
}
